python/beeai_framework/adapters/mcp/serve/server.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2025 © BeeAI a Series of LF Projects, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections.abc import Callable, Coroutine
+from typing import Any, Generic, Literal
+
+from pydantic import BaseModel, Field
+from typing_extensions import TypeVar
+
+from beeai_framework.tools.tool import AnyTool, Tool
+from beeai_framework.tools.types import ToolOutput
+
+try:
+    import mcp.server.fastmcp.prompts as mcp_prompts
+    import mcp.server.fastmcp.resources as mcp_resources
+    import mcp.server.fastmcp.server as mcp_server
+except ModuleNotFoundError as e:
+    raise ModuleNotFoundError(
+        "Optional module [mcp] not found.\nRun 'pip install \"beeai-framework[mcp]\"' to install."
+    ) from e
+
+
+from beeai_framework.serve.server import Server
+from beeai_framework.utils import ModelLike
+from beeai_framework.utils.models import to_model
+
+TInput = TypeVar("TInput", bound=Any, default=Any)
+
+
+class McpServerConfig(BaseModel):
+    """Configuration for the McpServer."""
+
+    transport: Literal["stdio", "sse"] = "stdio"
+    name: str = "MCP Server"
+    instructions: str | None = None
+    settings: mcp_server.Settings[Any] = Field(default_factory=lambda: mcp_server.Settings())
+
+
+class McpServer(
+    Generic[TInput],
+    Server[
+        TInput,
+        mcp_prompts.Prompt | mcp_resources.Resource | Callable[[Any], ToolOutput],
```

Comment:
Extract to a type alias.

```python
McpServerTool = Callable[[Any], ToolOutput]
McpServerEntry = mcp_prompts.Prompt | mcp_resources.Resource | McpServerTool
```

One question though, do we support async callable? If so, then replace `Callable[[Any], ToolOutput]` with `MaybeAsync[[Any], ToolOutput]`